src/Parser/Warlock/Demonology/Modules/Features/Checklist.js

```diff
@@ -0,0 +1,158 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import CoreChecklist, { Requirement, Rule } from 'Parser/Core/Modules/Features/Checklist';
+import SpellLink from 'common/SpellLink';
+import {PreparationRule} from 'Parser/Core/Modules/Features/Checklist/Rules';
+import {GenericCastEfficiencyRequirement} from 'Parser/Core/Modules/Features/Checklist/Requirements';
+import CastEfficiency from 'Parser/Core/Modules/CastEfficiency';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import Abilities from 'Parser/Core/Modules/Abilities';
+import LegendaryUpgradeChecker from 'Parser/Core/Modules/Items/LegendaryUpgradeChecker';
+import PrePotion from 'Parser/Core/Modules/Items/PrePotion';
+import EnchantChecker from 'Parser/Core/Modules/Items/EnchantChecker';
+import SoulShardDetails from 'Parser/Warlock/Demonology/Modules/SoulShards/SoulShardDetails';
+import SoulShardTracker from 'Parser/Warlock/Demonology/Modules/SoulShards/SoulShardTracker';
+import AlwaysBeCasting from 'Parser/Warlock/Demonology/Modules/Features/AlwaysBeCasting';
+import LegendaryCountChecker from 'Parser/Core/Modules/Items/LegendaryCountChecker';
+import DoomguardInfernal from 'Parser/Warlock/Demonology/Modules/Features/DoomguardInfernal';
+import Felstorm from 'Parser/Warlock/Demonology/Modules/Features/Felstorm';
+import CallDreadstalkers from 'Parser/Warlock/Demonology/Modules/Features/CallDreadstalkers';
+import DEUptimePet from 'Parser/Warlock/Demonology/Modules/Features/DEUptimePet';
+import AbilityTracker from 'Parser/Core/Modules/Combatants';
+import DEUptimeDreadstalkers from 'Parser/Warlock/Demonology/Modules/Features/DEUptimeDreadstalkers';
+import DEUptimeHoG from 'Parser/Warlock/Demonology/Modules/Features/DEUptimeHoG';
+import DEUptimeDoomguardInfernal from 'Parser/Warlock/Demonology/Modules/Features/DEUptimeDoomguardInfernal';
+
+
+class Checklist extends CoreChecklist{
+  static dependencies = {
+    abilities: Abilities,
+    abilityTracker: AbilityTracker,
+    combatants : Combatants,
+    castEfficiency: CastEfficiency,
+    alwaysBeCasting: AlwaysBeCasting,
+    legendaryUpgradeChecker: LegendaryUpgradeChecker,
+    legendaryCountChecker: LegendaryCountChecker,
+    prePotion: PrePotion,
+    doomguardInfernal : DoomguardInfernal,
+    felstorm: Felstorm,
+    callDreadstalkers: CallDreadstalkers,
+    deUptimePet: DEUptimePet,
+    deUptimeDreadstalkers: DEUptimeDreadstalkers,
+    deUptimeHog: DEUptimeHoG,
+    deUptimeDoomguardInfernal: DEUptimeDoomguardInfernal,
+    enchantChecker: EnchantChecker,
+    soulShardDetails: SoulShardDetails,
+    soulShardTracker: SoulShardTracker,
+  };
+
+
+  rules = [
```

Comment:
Your base (rotation spells) rule looks kinda incomplete, is this all that's important for Demonology Warlocks? I've never played the spec so I don't know but going through a guide how about including the following things:
- Uptime on Doom
- The cast efficiency on the artifact ability (Thal'kiel's Consumption)
- The cast efficiency on Summon Doomguard
- Summon Darkglare cast efficiency
- Grimoire: Felguard cast efficiency
- Call Dreadstalkers cast efficiency
- Hand of Gul'dan cast efficiency with 4 shards (probably needs a custom module)
- Soul Harvest cast efficiency
- Filler usage: Demonwrath vs Life Tap.

Reply:
Doom uptime is a good idea to add. Cast efficiency on consumption is iffy and would need more work since you usually want to delay it until you have a minimum number of demons active. The Doomguard, Darkglare, and Grimoire: Felguard cast efficiencies are already present in the offensive CD rule. Call Dreadstalkers cast efficiency is already present. The last 3 points are good ideas to add that I will work on.